spec/formula_spec.rb

```diff
@@ -0,0 +1,52 @@
+require 'spec_helper.rb'
+
+describe Statsample::GLM::Formula do
+  context '#parse_formula' do
+    context 'no interaction' do
+      let(:formula) { described_class.new 'y ~ a+b' }
+      subject { formula.parse_formula }
+      
+      it { is_expected.to be_a Array }
+      its(:first) { is_expected.to eq Statsample::GLM::Token }
+      its(:to_basic_formula) { is_expected.to eq 'y ~ 1+a(-)+b(-)' }
```

Comment:
Hmm. It becames a bit mangled here (at least for me). Fix me if I'm wrong:
* `Formula#initialize`, in fact, _parses_ the formula (into left part and tokens of right part);
* `Formula#parse_formula`, in fact, converts formula to some "canonical" set of tokens (and immediately returns them)
* For testing, we want string representation of this set, so `#parse_formula` has an option "return string representation of canonical set of tokens";
* `#parse_formula` will perform "canonicalization" on each call, yet result will always be the same. 

Is it accurate description?..

If it is, what do you think of this kind of design:
* `Formula#initialize` parses formula (storing original flow of tokens in `@tokens`) and canonicalizes it (storing tokens in `@canonical_tokens`);
* `Formula#canonical_to_s` renders canonical formula string
* Wherever you need this canonical tokens, you use formula, not "naked" array. For ex., [here](https://github.com/SciRuby/statsample-glm/pull/31/files#diff-1cb2585079e0550dd794342f9b6cc08dR11) could be written something like
```ruby
@formula.process_dataframe(@df)
```
...or something like this.

Better incapsulation, it seems.

NB: I use the word "canonical"/"canonicalize", though I'm absolutely not sure it is good term for what's going on in this formula processing :)